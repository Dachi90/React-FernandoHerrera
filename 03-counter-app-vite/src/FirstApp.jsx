export const FirstApp = () => {
	return <h1>Fernando</h1>;
};
