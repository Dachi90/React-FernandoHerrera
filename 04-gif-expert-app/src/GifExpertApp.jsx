import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch', 'Dragon ball']);
	const onAddCategory = () => {
		//setCategories([...categories, 'Super Campeones']);
		setCategories(['Super Campeones', ...categories]);
	};

	return (
		<>
			{/* t√≠tulo */}
			<h1>GifExpertApp</h1>

			{/* Input */}
			<AddCategory />

			{/* Listado de Gifs */}
			<button onClick={onAddCategory}>Agregar</button>
			<ol>
				{categories.map((category) => {
					return <li key={category}>{category}</li>;
				})}
			</ol>
			{/* Gif Item */}
		</>
	);
};
