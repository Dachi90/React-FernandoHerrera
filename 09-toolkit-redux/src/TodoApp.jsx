import { useGetTodosQuery } from './store/apis';

export const TodoApp = () => {
	const { data: todos = [], isLoading } = useGetTodosQuery();

	return (
		<>
			<h1>Todos - RTK Query</h1>
			<hr />
			<h4>IsLoading {isLoading ? 'true' : 'false'}</h4>

			<pre>...</pre>

			<ul>
				{todos.map((todo) => (
					<li key={todo.id}>
						<strong>{todo.completed ? 'DONE ' : 'PENDING '}</strong>
						{todo.title}
					</li>
				))}
			</ul>

			<button>Next TODO</button>
		</>
	);
};
