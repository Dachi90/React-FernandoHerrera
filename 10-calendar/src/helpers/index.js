export * from './calendarLocalizer';
export * from './convertEventesToDateEvents';
export * from './getEnvVariables';
export * from './getMessages';
