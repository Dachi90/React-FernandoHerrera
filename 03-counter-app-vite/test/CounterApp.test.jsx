import { render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas en el componente <CounterApp />', () => {
	const value = 100;
	test('Debe de hacer match con el snapshot', () => {
		const { containter } = render(<CounterApp value={value} />);
		expect(containter).toMatchSnapshot();
	});

	test('El valor de "value" debe ser 100', () => {
		render(<CounterApp value={value} />);
		expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100');
	});
});
