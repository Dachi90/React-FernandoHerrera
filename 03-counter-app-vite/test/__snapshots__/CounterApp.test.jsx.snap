// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp /> Debe de hacer match con el snapshot 1`] = `undefined`;
